presto-hive/src/test/java/com/facebook/presto/hive/parquet/ParquetTester.java

```diff
@@ -171,49 +220,153 @@ public void assertRoundTrip(ObjectInspector objectInspector, Iterable<?> writeVa
         }
     }
 
-    public void assertRoundTrip(ObjectInspector objectInspector, Iterable<?> writeValues, Iterable<?> readValues, Type type)
-            throws Exception
-    {
-        assertRoundTrip(objectInspector, writeValues, readValues, type, Optional.empty());
-    }
-
-    private static void assertFileContents(JobConf jobConf,
-            TempFile tempFile,
+    private static void assertFileContents(TempFile tempFile,
```

Comment:
Just curious what is the reason behind this change, what was the previous version missing? 

Also what you do here looks like what we already do in `TestHiveFileFormats::testParquetPageSource()`, so please try to reuse the methods already available instead of handling the decode logic yourself here. e.g., check `TestHiveFileFormats::assertRead()` to see `testPageSourceFactory`/`testCursorProvider` methods.

Reply:
The previous version
1) was reading primitives only
2) was reading blocks directly, not pages/cursor
One of the bugs optimised parquet reader has is related to reading complex structs from multipage parquet files. When an array's "head" at one page and it's "tail" on the next page,   offsets are calculated incorrectly(such array is treated as two arrays). So from second page column with arrays has shifted values. On the last page user gets an exception, because column with array and column with primitives have different number of rows.

I needed to test this fix, that's why switched to by page reading. Old reader(RecordCursor) is not really necessary here. I just added it to be able to test both readers with the same tests.